airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultStateAggregator.java

```diff
@@ -0,0 +1,47 @@
+package io.airbyte.workers.internal;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.config.State;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+
+public class DefaultStateAggregator implements StateAggregator {
+
+  private AirbyteStateType stateType = null;
+  private final StateAggregator streamStateAggregator = new StreamStateAggregator();
+  private final StateAggregator singleStateAggregator = new SingleStateAggregator();
+
+  @Override public void ingest(final AirbyteStateMessage stateMessage) {
+    checkTypeOrSetType(stateMessage.getType());
+
+    getStateAggregator().ingest(stateMessage);
```

Comment:
OPT: If we have no explicit use case for the `StreamStateAggregator` and `SingleStateAggregator`, I'd say the `getStateAggregator()` logic feels a bit heavy. 
That split makes the logic heavier since we have a checkOrSet and again a check in the nested object.
Not feeling super strongly about this but I would expect in the end an `if (STREAM) { ... } else { ... }` to be easier to read.

Reply:
The split was done to avoid having member that will be storing data but unused (stream have a map, single have just an object). The organization goal was to be close to what have been done for `DestStateLifecycleManager`

Reply:
I remove the check in the nested object because now its scope doesn't allow to use it outside of this class. @gosusnp Does it looks better now.